### PR TITLE
Handle 405 response

### PR DESCRIPTION
**Pull request by nadiacollado**
- Handles 405 response for requests with not allowed methods
- Replaces `isRequestValid()` with `isPathValid()` and `isMethodValid()`
- Update `Constants` class to include ALLOW variable

### CODE REVIEW

File: src/main/java/httpserver/response/ResponseWriter.java [hunk 1]
```diff
@@ -13,13 +13,13 @@ public Response buildSuccessResponse(Request request, String[] methods) {
         if (request.path.equals("/simple_get_with_body")) {
             response = new ResponseBuilder()
                     .setStatusCode(StatusCodes.SUCCESS)
-                    .addHeader("Allow", methods)
+                    .addHeader(Constants.ALLOW, methods)
```
Comment (ranizilpelwar) [inline src/main/java/httpserver/response/ResponseWriter.java:16]: Nice!

File: src/main/java/httpserver/router/Router.java [hunk 2]
```diff
@@ -19,25 +19,26 @@ public Router() {
...
+    }
+
+    public boolean isPathValid(Request request) {
+        return routes.containsKey(request.path);
+    }
+
+    public boolean isMethodValid(Request request, String[] methods) {
+        for (String method : methods) {
+            if (method.equals(request.method)) {
+                return true;
```
Comment (ranizilpelwar) [inline src/main/java/httpserver/router/Router.java:39]: Glad this worked out and nice use of last PR comment of for loop syntax on this PR!

File: src/test/java/httpserver/router/RouterTest.java [hunk 3]
```diff
@@ -41,4 +30,38 @@ void returnsRoutesForGivenRequest() {
 
         assertEquals("[GET, HEAD]", methods);
     }
+
+    @Test
+    void returnsTrueIfPathExistsInRouter() {
```
Comment (ranizilpelwar) [inline src/test/java/httpserver/router/RouterTest.java:35]: Yay tests!

File: src/test/java/httpserver/router/RouterTest.java [hunk 4]
```diff
@@ -41,4 +30,38 @@ void returnsRoutesForGivenRequest() {
...
+    void returnsTrueIfMethodIsAllowedForRequest() {
+        Request request = new Request(Methods.GET,"/simple_get", Constants.PROTOCOL, null, null);
+        Router router = new Router();
+        String[] methods = router.getMethods(request);
+
+        assertTrue(router.isMethodValid(request, methods));
+    }
+
+    @Test
+    void returnsFalseIfMethodIsNotAllowedForRequest() {
```
Comment (ranizilpelwar) [inline src/test/java/httpserver/router/RouterTest.java:60]: I try to include both positive and negative tests to ensure the logic is sound, as you did here. Nice work.

**Review by ranizilpelwar — APPROVED** [review]

File: src/test/java/httpserver/router/RouterTest.java (hunk 4 above)
Comment (nadiacollado) [reply src/test/java/httpserver/router/RouterTest.java:60]: Learned from the best. Thanks, Rani!

**Pull request merged**